model/generate.go

```diff
@@ -270,6 +256,29 @@ func (g *GeneratedProject) saveNewBuildsAndTasks(ctx context.Context, cachedProj
 	if err := AddNewBuilds(ctx, true, v, p, newTVPairsForNewVariants, g.TaskID); err != nil {
 		return errors.Wrap(err, "errors adding new builds")
 	}
+
+	newTasks := append(newTVPairsForExistingVariants.ExecTasks, newTVPairsForNewVariants.ExecTasks...)
+	if err = addDependencies(t, p, v, newTasks); err != nil {
+		return errors.Wrap(err, "error adding dependencies")
+	}
+
+	return nil
+}
+
+func addDependencies(t *task.Task, p *Project, v *Version, newTasks TVPairSet) error {
+	taskIDTable := NewTaskIdTable(p, v, "", "")
+	newDependencyIDs := []string{}
+	for _, newTask := range newTasks {
+		newDependencyIDs = append(newDependencyIDs, taskIDTable.ExecutionTasks.GetId(newTask.Variant, newTask.TaskName))
+	}
+
+	statuses := []string{evergreen.TaskSucceeded, task.AllStatuses}
```

Comment:
Are we intentionally not updating dependencies on a failed task?

Reply:
my thinking was that if the generator will have failed we won't get this far so we don't have to worry about updating dependencies for tasks that depend on the generator failing.